sklearn/model_selection/_search.py

```diff
@@ -819,8 +885,11 @@ def _store(key_name, array, weights=None, splits=False, rank=False):
                    splits=True, rank=True,
                    weights=test_sample_counts if iid else None)
             if self.return_train_score:
+                # _store('train_%s' % scorer_name, train_scores[scorer_name],
+                #        splits=True)
                 _store('train_%s' % scorer_name, train_scores[scorer_name],
-                       splits=True)
+                       splits=True,
+                       weights=train_sample_counts if iid else None)
```

Comment:
I'm not sure about this `iid` parameter.  I copied it from the `test_%s` case a few lines above.  I feel like you want it more when you don't have *IID*, so this is a little weird to me, but that's how it is above.

Reply:
I agree. But seems like putting correlation structure here is too complicated to implement